### DISCUSSION TITLE
Using forceDirected2d doesn't seem to have a "pulling and bouncing" effect

### DISCUSSION THREAD

**Discussion by FedericoCampe8** [reaviz/reagraph#105] [Q&A] [answered]
As per title, when I use forceDirected2d layout in my React project, it doesn't work like this demo:
[](https://observablehq.com/@d3/force-directed-graph)Force-Directed Graph

My code is simple:
`const ReaGraph = ({ nodes, edges }: GraphProps) => {
    return (
        <GraphCanvas
            nodes={nodes}
            edges={edges}
            theme={theme}
            layoutType={'forceDirected2d'}
            defaultNodeSize={3}
            draggable={true}
            edgeArrowPosition={'none'}
        />
    )
}`

Am I missing something?

**Comment by amcdnl** [comment]
I'm not sure what your asking. Can you provide more details please?

**Comment by FedericoCampe8** [comment]
Sure, let's take this 'draggable node' example:
[https://reagraph.dev/iframe.html?args=&id=demos-nodes--draggable](https://reagraph.dev/iframe.html?args=&id=demos-nodes--draggable)
Here we can pull a node that remains in the same x/y position as where it was pulled.

Now, consider this example
[https://observablehq.com/@d3/force-directed-graph](https://observablehq.com/@d3/force-directed-graph)
and pull any node. You see that the network re-adjusts and a force is applied to the node being pulled.

How can we obtain or, is it possible to obtain the same effect of the second example?

**Reply by amcdnl** [reply, marked as answer]
Oh - The force animation - I specifically did not add that. Its cool to see but its bad from a data science perspective - having nodes move when you move one is not a good ux for the user.

**Comment by FedericoCampe8** [comment]
THat makes sense to me. Thanks!
